nav2_costmap_2d/include/nav2_costmap_2d/nav2_3d_static_layer.hpp

```diff
@@ -0,0 +1,70 @@
+//
+// Created by sun on 2020/10/10.
+//
+
+#ifndef NAV2_3D_STATIC_LAYER_NAV2_3D_STATIC_LAYER_HPP
+#define NAV2_3D_STATIC_LAYER_NAV2_3D_STATIC_LAYER_HPP
+
+#include "nav2_costmap_2d/costmap_2d.hpp"
+#include "nav2_costmap_2d/costmap_layer.hpp"
+#include "nav2_costmap_2d/layer.hpp"
+#include "nav2_costmap_2d/layered_costmap.hpp"
+#include "nav2_costmap_2d/obstacle_layer.hpp"
+
+namespace nav2_costmap_2d  // TODO rename ns to nav2_costmap_2d
+{
+// TODO 3D static layer now inherits from obstacle layer
+// TODO but we have to discuss the overall design of 2D/3D data input and processing
+class Nav23dStaticLayer : public nav2_costmap_2d::ObstacleLayer
+{
+    /*
+     *
+     */
+public:
+    Nav23dStaticLayer(); // TODO a bit weird
+    virtual ~Nav23dStaticLayer();
+
+// TODO Confirm: Is the 3 methods declaration necessary here?
+// TODO Should 'override' keyword be placed here?
+// TODO Best practice： void onInitialize() override;
+// TODO What is exactly 'virtual' in C++?
+    virtual void onInitialize();
+    virtual void updateBounds(
+            double robot_x, double robot_y, double robot_yaw, double * min_x,
```

Comment:
We use 2 spaces in C++ Google / ROS2 styling

Reply:
Please run the linters: `ament_cpplint` and `ament_uncrustify`

Reply:
Thanks for your suggestion, I have tried ament_uncrustify for changing similar lints.